src/plugins/embeddable/public/lib/embeddable_saved_object_registry/embeddable_saved_object_registry.ts

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { PresentationContainer } from '@kbn/presentation-containers';
+import { FinderAttributes, SavedObjectCommon } from '@kbn/saved-objects-finder-plugin/common';
+import { SavedObjectMetaData } from '@kbn/saved-objects-finder-plugin/public';
+
+type SOToEmbeddable<TSavedObjectAttributes extends FinderAttributes = FinderAttributes> = (
+  container: PresentationContainer,
+  savedObject: SavedObjectCommon<TSavedObjectAttributes>
+) => void;
+
+export type ReactEmbeddableSavedObject<
+  TSavedObjectAttributes extends FinderAttributes = FinderAttributes
+> = {
+  method: SOToEmbeddable<TSavedObjectAttributes>;
+  savedObjectMetaData: SavedObjectMetaData;
+};
+
+const registry: Map<string, ReactEmbeddableSavedObject<any>> = new Map();
+
+export const registerReactEmbeddableSavedObject = <TSavedObjectAttributes extends FinderAttributes>(
+  type: string,
+  method: SOToEmbeddable<TSavedObjectAttributes>,
```

Comment:
How about a better name then method. Maybe something like `onAdd` or something like that.